src/main/java/br/com/jtech/services/daily/manager/application/core/usecases/daily/CreateDailyUseCase.java

```diff
@@ -1,36 +1,86 @@
+ *
+ */
 package br.com.jtech.services.daily.manager.application.core.usecases.daily;
 
 
 import br.com.jtech.services.daily.manager.application.core.domains.daily.Daily;
+import br.com.jtech.services.daily.manager.application.core.domains.employee.Employee;
+import br.com.jtech.services.daily.manager.application.core.domains.squad.Squad;
+import br.com.jtech.services.daily.manager.application.core.exceptions.employee.EmployeeBadRequestException;
+import br.com.jtech.services.daily.manager.application.core.exceptions.employee.EmployeeNotFoundException;
+import br.com.jtech.services.daily.manager.application.core.exceptions.squad.SquadNotFoundException;
 import br.com.jtech.services.daily.manager.application.ports.input.daily.CreateDailyInputGateway;
 import br.com.jtech.services.daily.manager.application.ports.output.daily.CreateDailyOutputGateway;
+import br.com.jtech.services.daily.manager.application.ports.output.employee.FindEmployeeByUsernameOutputGateway;
+import br.com.jtech.services.daily.manager.application.ports.output.squad.FindSquadByIdOutputGateway;
+import br.com.jtech.services.daily.manager.config.infra.utils.GenId;
+
+import java.util.Optional;
 
-/**
-* class ManagerUseCase  
-* 
-* user angelo.vicente@jtech.corp  
-*/
 public class CreateDailyUseCase implements CreateDailyInputGateway {
 
     private final CreateDailyOutputGateway createDailyOutputGateway;
 
-    public CreateDailyUseCase(CreateDailyOutputGateway createDailyOutputGateway) {
+    private final FindEmployeeByUsernameOutputGateway findEmployeeByUsernameOutputGateway;
+
+    private final FindSquadByIdOutputGateway findSquadByIdOutputGateway;
+
+    public CreateDailyUseCase(CreateDailyOutputGateway createDailyOutputGateway, FindEmployeeByUsernameOutputGateway findEmployeeByUsernameOutputGateway, FindSquadByIdOutputGateway findSquadByIdOutputGateway) {
         this.createDailyOutputGateway = createDailyOutputGateway;
-     }
+        this.findEmployeeByUsernameOutputGateway = findEmployeeByUsernameOutputGateway;
+        this.findSquadByIdOutputGateway = findSquadByIdOutputGateway;
+    }
 
-    public Daily create(Daily daily) {
+    public Optional<Daily> create(Daily daily) {
+        var employee = getEmployee(daily.getAuthor());
+        var squad = getSquad(daily.getSquad());
+        daily.setAuthor(employee);
+        daily.setSquad(squad);
+        validateTasks(daily);
         return createDailyOutputGateway.create(daily);
-     }
- }
\ No newline at end of file
+    }
+
+    private void validateTasks(Daily daily) {
+        if (daily.getTasks() != null && !daily.getTasks().isEmpty()) {
+            for (var task : daily.getTasks()) {
+                var assignee = getEmployee(task.getAssignee());
+                task.setAssignee(assignee);
+                if (task.getId() == null) {
+                    task.setId(GenId.newUuid());
+                }
+            }
+        }
+    }
+
+    private Employee getEmployee(Employee employee) {
```

Comment:
O código não está errado... mas vamos abusar das novas features do Java

private Employee getEmployee(Employee employee) {
    Objects.requireNonNull(employee, "Employee is null!");
    String username = Optional.ofNullable(employee.getUsername())
                              .orElseThrow(() -> new EmployeeBadRequestException("Username is invalid!"));

    return findEmployeeByUsernameOutputGateway.findByUsername(username)
                                               .orElseThrow(() -> new EmployeeNotFoundException("Employee not found!"));
}